doc/developers/maintainer.rst

```diff
@@ -17,61 +30,95 @@ Before a release
 
    - The ``maint_tools/whats_missing.sh`` script may be used to identify pull
      requests that were merged but likely missing from What's New.
 
-Preparing a bug-fix-release
-...........................
+4. Make sure the deprecations, FIXME and TODOs tagged for the release have
+   been taken care of.
+
+**Permissions**
+
+The release manager requires a set of permissions on top of the usual
+permissions given to maintainers, which includes:
+
+- *maintainer* role on ``scikit-learn`` projects on ``pypi.org`` and
+  ``test.pypi.org``, separately.
+- become a member of the *scikit-learn* team on conda-forge by editing the 
+  ``recipe/meta.yaml`` file on 
+  ``https://github.com/conda-forge/scikit-learn-feedstock``
+- *maintainer* on ``https://github.com/MacPython/scikit-learn-wheels``
+
+
+.. _preparing_a_release_pr:
 
-Since any commits to a released branch (e.g. 0.999.X) will automatically update
-the web site documentation, it is best to develop a bug-fix release with a pull
-request in which 0.999.X is the base. It also allows you to keep track of any
-tasks towards release with a TO DO list.
+Preparing a release PR
+......................
 
-Most development of the bug fix release, and its documentation, should
-happen in master to avoid asynchrony. To select commits from master for use in
-the bug fix (version 0.999.3), you can use::
+Releasing the first RC of e.g. version `0.99` involves creating the release
+branch `0.99.X` directly on the main repo, where `X` really is the letter X,
+**not a placeholder**. This is considered the *feature freeze*. The
+development for the major and minor releases of 0.99 should
+**also** happen under `0.99.X`. Each release (rc, major, or minor) is a tag
+under that branch.
 
+In terms of including changes, the first RC ideally counts as a *feature
+freeze*. Each coming release candidate and the final release afterwards will
+include minor documentation changes and bug fixes. Any major enhancement or
+feature should be excluded.
+
+The minor releases should include bug fixes and some relevant documentation
+changes only. Any PR resulting in a behavior change which is not a bug fix
+should be excluded.
+
+First, create a branch, **on your own fork** (to release e.g. `0.999.3`)::
+
+    $ # assuming master and upstream/master are the same
     $ git checkout -b release-0.999.3 master
-    $ git rebase -i 0.999.X
 
-Then pick the commits for release and resolve any issues, and create a pull
-request with 0.999.X as base. Add a commit updating ``sklearn.__version__``.
-Additional commits can be cherry-picked into the ``release-0.999.3`` branch
-while preparing the release.
+Then, create a PR **to the** `scikit-learn/0.999.X` **branch** (not to
+master!) with all the desired changes::
+
+	$ git rebase -i upstream/0.999.2
+
+It's nice to have a copy of the ``git rebase -i`` log in the PR to help others
+understand what's included.
 
 Making a release
-----------------
+................
```

Comment:
At what point do we merge the PR mentioned above into the 0.99.X branch? Should we add a step in the list below?

Reply:
It has to happen before tagging. You can add one if you think it's not clear.